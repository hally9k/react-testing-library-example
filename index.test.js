import * as React from 'react'
import { render } from 'react-testing-library'

describe('Input > Text', () => {
	it('should render changes', () => {
		const mockOnChange = jest.fn()
		const { getByTestId } = render(
			<input type="text" value="hello" onChange={mockOnChange} data-testid="my-input" />
		)

		expect(getByTestId('my-input')).toMatchSnapshot()

		getByTestId('my-input').value = 'yo!'

		expect(getByTestId('my-input')).toMatchSnapshot()
		expect(mockOnChange).toBeCalled()
	})
})
