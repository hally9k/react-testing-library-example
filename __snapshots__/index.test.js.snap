// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input > Text should render changes 1`] = `
<input
  data-testid="my-input"
  type="text"
  value="hello"
/>
`;

exports[`Input > Text should render changes 2`] = `
<input
  data-testid="my-input"
  type="text"
  value="hello"
/>
`;
